github/getRepo.go

```diff
@@ -0,0 +1,79 @@
+package github
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+
+	"gopkg.in/src-d/go-git.v4/plumbing"
+
+	"gopkg.in/src-d/go-billy.v4"
+
+	gHttp "gopkg.in/src-d/go-git.v4/plumbing/transport/http"
+
+	"github.com/google/go-github/github"
+	"golang.org/x/oauth2"
+	"gopkg.in/src-d/go-billy.v4/memfs"
+	git "gopkg.in/src-d/go-git.v4"
+	"gopkg.in/src-d/go-git.v4/storage/memory"
+)
+
+const (
+	tokenEnvVar = "PERSONAL_ACCESS_TOKEN"
+)
+
+// GetRepo returns an in memory filesystem with commit checked out
+func GetRepo(apiURL, org, name, headRef, baseRef string) (fs billy.Filesystem, changedDirs []string, err error) {
+	token, present := os.LookupEnv(tokenEnvVar)
+	if !present {
+		log.Fatalf("Environment variable %s is not exported.\n", tokenEnvVar)
+	}
+	context := context.Background()
+	tokenService := oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: token},
+	)
+	tokenClient := oauth2.NewClient(context, tokenService)
+	client, err := github.NewEnterpriseClient(apiURL, apiURL, tokenClient)
+	if err != nil {
+		err = fmt.Errorf("Cannot create github client: %v", err)
+		return
+	}
+
+	repo, _, err := client.Repositories.Get(context, org, name)
+	if err != nil {
+		err = fmt.Errorf("Cannot get github repo: %v", err)
+		return
+	}
+
+	fs = memfs.New()
+	url := repo.GetCloneURL()
+	r, err := git.CloneContext(context, memory.NewStorage(), fs, &git.CloneOptions{
+		URL:  url,
+		Auth: &gHttp.BasicAuth{Username: "none", Password: token},
+	})
+	if err != nil {
+		err = fmt.Errorf("Cannot clone github repo (%s): %v", url, err)
+		return
+	}
+
+	changedDirs, err = GetChangedProjects(r, headRef, baseRef)
```

Comment:
This feels like it doesn't really belong here - too many jobs at once.

Reply:
Great point, will refactor